cli/src/commands/config/set.rs

```diff
@@ -46,7 +47,85 @@ pub fn cmd_config_set(
             path = config_path.display()
         )));
     }
-    // TODO(#531): Infer types based on schema (w/ --type arg to override).
     let value = parse_toml_value_or_bare_string(&args.value);
+    // If the user is trying to change the author config, we should warn them that
+    // it won't affect the working copy author
+    if args.name == ConfigNamePathBuf::from_iter(vec!["user", "name"]) {
+        check_wc_user_name(command, ui, &value)?;
+    } else if args.name == ConfigNamePathBuf::from_iter(vec!["user", "email"]) {
+        check_wc_user_email(command, ui, &value)?;
+    };
+
     write_config_value_to_file(&args.name, value, &config_path)
 }
+
+/// Returns the commit of the working copy if it exists.
+fn maybe_wc_commit(helper: &WorkspaceCommandHelper) -> Option<Commit> {
+    let repo = helper.repo();
+    let maybe_wc_commit = helper
+        .get_wc_commit_id()
+        .map(|id| repo.store().get_commit(id))
+        .transpose()
+        .unwrap();
+    maybe_wc_commit
+}
+
+/// Check if the working copy author name matches the user's config value
+/// If it doesn't, print a warning message
+fn check_wc_user_name(
+    command: &CommandHelper,
+    ui: &mut Ui,
+    user_name: &toml_edit::Value,
+) -> Result<(), CommandError> {
+    let helper = command.workspace_helper(ui)?;
```

Comment:
you'll need to suppress error by `let Ok(helper) = .. else { return .. };` or something.

`config set` should work even if cwd isn't a jj repo.

Reply:
Done